src/react-freshchat.js

```diff
@@ -136,6 +136,15 @@ class FreshChat extends React.Component {
   componentWillUnmount() {
     widget().close()
   }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Does this fire on initial render as well? If so, there's probably no reason to still do the init in the constructor.

Reply:
https://cdn-images-1.medium.com/max/1600/0*VoYsN6eq7I_wjVV5.png

I think it is only available after the component has mounted so post the first render cycle

Reply:
correct, the issue here is if you change the props more than one time, it will call the freshchat init fn more than one time